api/cors.md

```diff
@@ -0,0 +1,19 @@
+
+
+### What defines an origin?
+
+The origin of a resource is defined by its URL. Specifically, the part of the URL before the first folder or file. Here are some examples of cross-origin and same-origin URLs:
+
+* http://foo.com and https://foo.com: **cross-origin** because of different [URI schemes](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier) (http/https)
+* http://www.foo.com and http://blog.foo.com: **cross-origin** because of different [hostnames](https://en.wikipedia.org/wiki/Hostname) (www/blog)
+* http://foo.com:3000 and http://foo.com:8080: **cross-origin** because of different [ports](https://en.wikipedia.org/wiki/Port_(computer_networking) (3000/8080)
+* http://foo.com and http://bar.com: **cross-origin** because of different domain names (foo/bar)
+* http://foo.com/profile and http://foo.com/v1/users: **same-origin**
+
+
+### Configuring CORS
+
+CORS can only be configured on the server that provides the restricted resource. That means if we want to consume an API from the browser, and the URL of the API doesn't have the same origin as the page our script is running in, the server must be configured to allow requests from other origins. When a web API server has been configured to accept requests from other origins, it will include an `Access-Control-Allow-Origin` HTTP response header. As long as our client-side script came from a matching origin, we shouldn't be effected by the same-origin security policy.
+
+But what if we can't change the configuration on the server? That just means we can't consume the API from a web browser. That doesn't mean we can't consume the API from our server. We just need to create an API endpoint for our client-side code to consume. In our API we simply call the external API that doesn't permit cross-origin requests. But isn't our web server a different origin than the external API? Yes, but the same-origin security policy is a browser security measure that prevents scripts from different origins from using the DOM to access information it shouldn't. We don't have to worry about that on a web server since there isn't a DOM.
```

Comment:
> We just need to create an API endpoint for our client-side code to consume
I had to read this twice to understand what you mean.

I think what it might be simpler to say is : 
We can route requests from our client via our server (we won't need a CORS setup if our client and server have the same origin) and get our server to hit other servers. Note that our server doesn't have same-origin constraint, because the primary reason for a same-origin constraint is to avoid script acting within a browser to read or interfere with the brwoser's DOM.

Or something like that. I think a diagram would make this quite clear.